scripts/generate/public_source_gen.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Copyright (c) F5, Inc.
+# This source code is licensed under the Apache License, Version 2.0 license found in the
+# LICENSE file in the root directory of this source tree.
+
+# shellcheck shell=bash
+# Generate support file from a public git repo
+# Only call it through //go:generate comments in project root to update .gen.go files
+
+set -e
+
+branch="master"
+url=""
+output_path=""
+sub_path=""
+config_path=""
+
+help() {
+    cat << EOF
+Generate support file from a public repo(currently only git).
+Only call me through //go:generate comments in project root to update .gen.go files.
+
+usage: $(basename "$0") [-b|--branch] [-o|--output-path] [-c|--config-path] [-s|--sub-path] [--url] [-h|--help]
+    -h  | --help                Display this message
+    -b  | --branch              Branch to checkout, defaults to "$branch". (optional)
+    -o  | --output-path         The path of the output file. (required)
+    -c  | --config-path         The path of json config file. Normally the json config file should be under nginx-go-crossplane/scripts/generate/configs.
+     The json config will be unmarshed into generator.GenerateConfig(at nginx-go-crossplane/internal/generator.go).
+     For the json config format, please refer to examples under nginx-go-crossplane/scripts/generate/configs,
+     or see generator.GenerateConfig directly. (required)
+    -s  | --sub-path            The subfolder or file within the repository contains all the directives you want, whole repository by default. (optional)
```

Comment:
Nit "folder" is a windows/UI concept. Use "directory".

I think this looks better:

```suggestion
    -p  | --path            Path to a directory of file in the repository containing
                            an nginx module. (optional)
```

Reply:
Applied. But a bit curious: is it okay we don't specify the default behavior here (scan the whole cloned repo)?

Reply:
I might specify the behavior in the main body of `help()` and keep the description of the flags short.

Reply:
Applied: move the default behavior to `help()`